src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipelineOptions.cs

```diff
@@ -3,77 +3,138 @@
 
 using Azure.Base.Http.Pipeline;
 using System;
-using System.Buffers;
+using System.Collections.Generic;
 using System.ComponentModel;
-using System.Diagnostics;
-using System.Threading.Tasks;
 
 namespace Azure.Base.Http
 {
-    public partial struct HttpPipeline
+    public class HttpPipelineOptions
     {
-        public class Options
-        {
-            static readonly HttpPipelinePolicy s_default = new Default();
-
-            HttpPipelineTransport _transport;
+        ServiceContainer _container = new ServiceContainer();
 
-            public ArrayPool<byte> Pool { get; set; } = ArrayPool<byte>.Shared;
+        HttpPipelineTransport _transport;
+        List<HttpPipelinePolicy> _perCallPolicies;
+        List<HttpPipelinePolicy> _perRetryPolicies;
 
-            public HttpPipelineTransport Transport {
-                get => _transport;
-                set {
-                    if (value == null) throw new ArgumentNullException(nameof(value));
-                    _transport = value;
-                }
+        public HttpPipelineTransport Transport {
+            get => _transport;
+            set {
+                if (value == null) throw new ArgumentNullException(nameof(value));
+                _transport = value;
             }
+        }
 
-            public HttpPipelinePolicy TelemetryPolicy { get; set; } = s_default;
+        public HttpPipelinePolicy LoggingPolicy { get; set; }
 
-            public HttpPipelinePolicy LoggingPolicy { get; set; } = s_default;
+        public HttpPipelinePolicy RetryPolicy { get; set; }
 
-            public HttpPipelinePolicy RetryPolicy { get; set; } = s_default;
+        public bool DisableTelemetry { get; set; } = false;
```

Comment:
What telemetry does this setting control? Metrics for tracing? The UA string? Something else?

Reply:
UA header

Reply:
Never mind - I can read the source code :). I don't think that setting the UA string is what people think of when you say "telemetry". Especially once we have integrated tracing policies into the mix. Where we presumably will send more data to a location controlled by the app.

Reply:
Yes, and this is one of the reasons I think the current name of the property is the right name, i.e. we want to be free to change what we do for telemetry and this property just turns it (whatever we do) on and off. The fact that today we send the UA header might be provisional.